client/my-sites/domains/controller.jsx

```diff
@@ -321,25 +319,8 @@ const jetpackNoDomainsWarning = ( context, next ) => {
 	const isJetpack = isJetpackSite( state, siteId ) && ! isSiteAutomatedTransfer( state, siteId );
 
 	if ( siteId && isJetpack ) {
-		context.primary = (
-			<Main>
-				<PageViewTracker
-					path={ context.path.startsWith( '/domains/add' ) ? '/domains/add' : '/domains/manage' }
-					title="My Sites > Domains > No Domains On Jetpack"
-				/>
-				<EmptyContent
-					title={ translate( 'Domains are not available for this site.' ) }
-					line={ translate(
-						'You can only purchase domains for sites hosted on WordPress.com at this time.'
-					) }
-					action={ translate( 'View Plans' ) }
-					actionURL={ '/plans/' + getSelectedSiteSlug( state ) }
-				/>
-			</Main>
-		);
-
-		makeLayout( context, noop );
-		clientRender( context );
+		context.store.dispatch( setSelectedSiteId( null ) );
+		page.redirect( '/domains/manage' );
```

Comment:
Without clearing selectedSiteId state first this is what we see when switching to a jetpack site:

![Screenshot 2023-07-19 at 17-34-53 Domains ‹ Test Site — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/e03ac13a-db8e-4fd7-a82c-ad54620fc6c5)

The jetpack domain shouldn't be listed at all.

I think a better fix for this might actually be inside /domains/manage routes to just clear the selected site ID when it's not present in the url on load.